source/shared/entries.js

```diff
@@ -0,0 +1,38 @@
+import { getSharedArchiveManager } from "../library/buttercup.js";
+import { EntryFinder } from "../library/buttercupCore";
+
+export async function getMatchingEntriesForSearchTerm(term) {
+    const manager = getSharedArchiveManager();
+
+    const unlockedSources = manager.unlockedSources;
+    const lookup = unlockedSources.reduce(
+        (current, next) => ({
+            ...current,
+            [next.workspace.archive.id]: next.id
+        }),
+        {}
+    );
+    const archives = unlockedSources.map(source => source.workspace.archive);
+    const finder = new EntryFinder(archives);
+
+    return Promise.all(
+        finder.search("/" + term + "/").map(async result => {
```

Comment:
What's with `"/" + term + "/"`?

Reply:
Not sure that this function needs to be `async` though, as there's no `await` calls.

Reply:
I misunderstood the finder.search function. I thought it was supposed to take a regex expression. Anyway, I removed those slashes and it worked fine. I also removed the async and everything worked as expected!

Reply:
Great! Thanks @chkichkiparmparm!